### PR TITLE
simulate raft leader and transport. 

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @disksing

### CODE REVIEW

File: server/cluster_worker_test.go [hunk 1]
```diff
@@ -184,6 +250,41 @@ func (n *mockRaftNode) run(c *C) {
 	}
 }
 
+func (n *mockRaftNode) runRaft(c *C) {
+	for {
+		select {
+		case msg := <-n.raftMsgCh:
+			n.handleRaftMsg(c, msg)
+		}
```
Comment (disksing) [inline server/cluster_worker_test.go]: No need to `select`.

File: server/cluster_worker_test.go [hunk 2]
```diff
@@ -207,13 +308,8 @@ func (n *mockRaftNode) handleRequest(c *C, req *raft_cmdpb.RaftCommandRequest) *
...
-	//	1. Use leader check for region, we can control where leader is.
-	//	2. Simulate raft message transport, leader can send this request to
-	//		other peers directly.
-
-	_, ok = store.peers[req.Header.GetRegionId()]
+	regionID := req.Header.GetRegionId()
+	peer, ok := store.peers[regionID]
 	if !ok {
 		resp := newErrorCmdResponse(errors.New("region not found"))
 		resp.Header.Error.KeyNotInRegion = &errorpb.KeyNotInRegion{
```
Comment (disksing) [inline server/cluster_worker_test.go]: Error type should be `RegionNotFoundError`?

Comment (siddontang) [inline server/cluster_worker_test.go]: yes.

File: server/cluster_worker_test.go (hunk 1 above)
Comment (siddontang) [inline server/cluster_worker_test.go]: I will add close channel to quit this goroutine later.

**Comment by disksing** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @ngaut

**Comment by ngaut** [comment]
LGTM

**Pull request merged**